podpac/core/coordinates/test/test_array_coordinates1d.py

```diff
@@ -742,41 +486,66 @@ def test_index(self):
         with pytest.raises(IndexError):
             c[10]
 
-    def test_index_segment_lengths(self):
-        # array segment_lengths
-        c = ArrayCoordinates1d([1, 2, 4, 5], segment_lengths=[0.1, 0.2, 0.3, 0.4])
 
-        c2 = c[1]
-        assert c2.segment_lengths == 0.2 or np.array_equal(c2.segment_lengths, [0.2])
+class TestArrayCoordinatesAreaBounds(object):
+    def test_get_area_bounds_numerical(self):
+        values = np.array([0.0, 1.0, 4.0, 6.0])
+        c = ArrayCoordinates1d(values)
 
-        c2 = c[1:3]
-        assert_equal(c2.segment_lengths, [0.2, 0.3])
+        # point
+        area_bounds = c.get_area_bounds(None)
+        assert_equal(area_bounds, [0.0, 6.0])
 
-        c2 = c[[2, 1]]
-        assert_equal(c2.segment_lengths, [0.3, 0.2])
+        # uniform
+        area_bounds = c.get_area_bounds(0.5)
+        assert_equal(area_bounds, [-0.5, 6.5])
 
-        c2 = c[[]]
-        assert_equal(c2.segment_lengths, [])
+        # segment
+        area_bounds = c.get_area_bounds([-0.2, 0.7])
+        assert_equal(area_bounds, [-0.2, 6.7])
 
-        # uniform segment_lengths
-        c = ArrayCoordinates1d([1, 2, 4, 5], segment_lengths=0.5)
+        # polygon (i.e. there would be corresponding offets for another dimension)
+        area_bounds = c.get_area_bounds([-0.2, -0.5, 0.7, 0.5])
```

Comment:
I don't quite understand this one...

Reply:
I'm not sure if we will ever use this one, but it is very general.

For a centered square, you could use `boundary = {'lat': [-1.0, -1.0, 1.0, 1.0], 'lon': [-1.0, 1.0, -1.0, 1.0]}`. So for lat, you get boundary offsets `[-1.0, -1.0, 1.0, 1.0]`, which appear a bit meaningless at that point.

For area bounds, all you care about is the min and max offset, and currently Coordinates1d `get_area_bounds` does that conversion (from boundary offsets to area bounds offsets). I guess it could be more clear if full Coordinates `get_area_bounds` method did the conversion, e.g:

```
class Coordinates:
    def get_area_bounds(self, boundary):
        offsets = {}
        for dim, value in boundary.items():
            if np.ndarray(vaue).ndim == 0:
               value = make_coord_delta(value)
               offsets[dim] = [-value, value]
            elif np.ndarray(value).ndim == 1:
                value = make_coord_delta_array(value)
                offsets[dim] = [min(value), max(value)]
            else:
                value = np.array([make_coord_delta_array(elem) for elem in value])
                offsets[dim] = np.array(np.min(value, axis=1), np.max(value, axis=1)).T
                
        return {dim: self[dim].get_area_bounds(offsets.get(dim)) for dim in self.udims}
```

And then Coordinates1d.get_area_bounds would expect offsets, and be a bit clearer:

```
class Coordinates1d:
    def get_area_bounds(self, offsets):
        # point coordinates
        if offsets is None:
            return self.bounds

        # empty coordinates
        if self.size == 0:
            return self.bounds

        if np.array(offsets).ndim == 1:
            # uniform boundary
            lo_offset, hi_offset = offsets
        else:
            # non-uniform boundary
            L, H = self.argbounds
            lo_offset = offsets[L][0]
            hi_offset = offsets[H][1]

        lo, hi = self.bounds
        lo = add_coord(lo, lo_offset)
        hi = add_coord(hi, hi_offset)

        return lo, hi
```

That is a little bit less efficient, but maybe worth it.

Reply:
I'm not sure when/where we will need the area_bounds though.

Reply:
Ah, this is great for future. Nice. Good spec.